### PR TITLE
fix(cucumberjs): attach a datable to the step

### PR DESCRIPTION
**Pull request by egerix**
### Context
split pr https://github.com/allure-framework/allure-js/pull/546
When I switched to allure in cucumber v8, I encountered a number of problems that were not repeated in cucumber v6:
- Attachments with the datatable are placed at the end of the test, not under the expected step. This is very unattractive if the test has several datatables. Previously, in Cucumber 6, they were placed under each step.

some pictures with changes:
**problem before fix:**
image

**problem after fix:**
image

#### Checklist
- [x] [Sign Allure CLA][cla]
- [x] Provide unit tests

[cla]: https://cla-assistant.io/accept/allure-framework/allure2

### CODE REVIEW

File: packages/allure-cucumberjs/src/CucumberJSAllureReporter.ts [hunk 1]
```diff
@@ -581,6 +559,26 @@ export class CucumberJSAllureFormatter extends Formatter {
...
+
+      if (!argument?.dataTable) {
+        return;
+      }
+
+      const csvDataTable = argument.dataTable.rows.reduce(
+        (acc, row) => `${acc + row.cells.map((cell) => cell.value).join(",")}\n`,
+        "",
+      );
+      const attachmentFilename = this.allureRuntime.writeAttachment(csvDataTable, "text/csv");
```
Comment (vovsemenv) [inline packages/allure-cucumberjs/src/CucumberJSAllureReporter.ts:573]: Consider using `ContentType.CSV`

File: packages/allure-cucumberjs/src/CucumberJSAllureReporter.ts [hunk 2]
```diff
@@ -581,6 +559,26 @@ export class CucumberJSAllureFormatter extends Formatter {
...
+
+      allureStep.addAttachment(
+        "Data table",
+        {
+          contentType: "text/csv",
```
Comment (vovsemenv) [inline packages/allure-cucumberjs/src/CucumberJSAllureReporter.ts:578]: `ContentType.CSV`

File: packages/allure-cucumberjs/src/CucumberJSAllureReporter.ts [hunk 3]
```diff
@@ -581,6 +559,26 @@ export class CucumberJSAllureFormatter extends Formatter {
           .find((kw) => kw !== undefined) || "";
       const allureStep = currentTest.startStep(keyword + ps.text, Date.now());
       this.allureSteps.set(data.testStepId, allureStep);
+
+      const {argument} = ps;
```
Comment (vovsemenv) [inline packages/allure-cucumberjs/src/CucumberJSAllureReporter.ts:563]: please fix formatting
`const { argument } = ps;`

**Review by vovsemenv — COMMENTED** [review]
Looks good overall, please fix formatting

File: packages/allure-cucumberjs/src/CucumberJSAllureReporter.ts (hunk 1 above)
Comment (egerix) [reply packages/allure-cucumberjs/src/CucumberJSAllureReporter.ts]: fixed 1afd41a1fa503604bfca359bf208a4fbe0e0d8e7

File: packages/allure-cucumberjs/src/CucumberJSAllureReporter.ts (hunk 2 above)
Comment (egerix) [reply packages/allure-cucumberjs/src/CucumberJSAllureReporter.ts]: fixed 1afd41a1fa503604bfca359bf208a4fbe0e0d8e7

File: packages/allure-cucumberjs/src/CucumberJSAllureReporter.ts (hunk 3 above)
Comment (egerix) [reply packages/allure-cucumberjs/src/CucumberJSAllureReporter.ts]: fixed 1afd41a1fa503604bfca359bf208a4fbe0e0d8e7

**Review by baev — APPROVED** [review]

**Comment by baev** [comment]
@egerix thanks! 🎉

**Pull request merged**